server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -640,6 +653,54 @@ private void checkAliasNameConflicts(Map<String, String> renamedIndices, Set<Str
 
+                    private String[] getIgnoreSettingsInternal() {
+                        // for non-remote store enabled domain, we will remove all the remote store
+                        // related index settings present in the snapshot.
+                        String[] indexSettingsToBeIgnored = new String[] {};
+                        if (false == RemoteStoreNodeAttribute.isRemoteStoreAttributePresent(clusterService.getSettings())) {
+                            indexSettingsToBeIgnored = ArrayUtils.concat(indexSettingsToBeIgnored, new String[] { "index.remote_store.*" });
+                        }
+                        return indexSettingsToBeIgnored;
+                    }
+
+                    private Settings getOverrideSettingsInternal() {
+                        final Settings.Builder settingsBuilder = Settings.builder();
+                        updateReplicationStrategy(
+                            settingsBuilder,
+                            request.indexSettings(),
+                            clusterService.getSettings(),
+                            clusterService.getClusterSettings()
+                        );
+                        // remote store settings needs to be overridden in the cluster where snapshot is being restored if:
+                        // 1. the remote store feature is enabled
+                        // 2. cluster is undergoing remote store migration
+                        MetadataCreateIndexService.updateRemoteStoreSettings(settingsBuilder, clusterService.getSettings());
+                        MetadataCreateIndexService.updateRemoteStoreSettingsForMigration(
+                            settingsBuilder,
+                            clusterService.state(),
+                            clusterService.getClusterSettings()
+                        );
+                        return settingsBuilder.build();
+                    }
+
+                    private void updateReplicationStrategy(
+                        Settings.Builder settingsBuilder,
+                        Settings requestSettings,
+                        Settings nodeSettings,
+                        ClusterSettings clusterSettings
+                    ) {
+                        // if remote store feature is enabled, override replication strategy to segment.
+                        // if `cluster.index.restrict.replication.type` is enabled and user have not provided
+                        // replication strategy setting with restore request, override replication strategy
+                        // setting to cluster default.
+                        if (isRemoteStoreAttributePresent(nodeSettings)) {
+                            settingsBuilder.put(SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT);
+                        } else if (!INDEX_REPLICATION_TYPE_SETTING.exists(requestSettings)
+                            && clusterSettings.get(IndicesService.CLUSTER_INDEX_RESTRICT_REPLICATION_TYPE_SETTING)) {
```

Comment:
if index is in migration and direction is remote store, can the `CLUSTER_INDEX_RESTRICT_REPLICATION_TYPE_SETTING` be true and `CLUSTER_REPLICATION_TYPE_SETTING` be `DOCUMENT` ?

Reply:
Deferred `RestoreService.java` changes for now.